client/src/app/containers/Game/CountAction.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { useState } from 'react';
+
+import Button from '@material-ui/core/Button';
+import FormControl from '@material-ui/core/FormControl';
+import FormGroup from '@material-ui/core/FormGroup';
+import Input from '@material-ui/core/Input';
+import InputLabel from '@material-ui/core/InputLabel';
+import SendIcon from '@material-ui/icons/Send';
+
+import { ActionInputProps } from './types';
+import { useGame } from './useGame';
+
+interface Props extends ActionInputProps {
+    isCrib: boolean;
+}
+
+const CountAction: React.FunctionComponent<Props> = ({
+    isBlocking,
+    isCrib,
+}) => {
+    const [points, setPoints] = useState(0);
+    const { submitCountHandAction, submitCountCribAction } = useGame();
+
+    const submitAction = isCrib ? submitCountCribAction : submitCountHandAction;
+    const componentId = isCrib ? 'count-crib-input' : 'count-hand-input';
+
+    return (
+        <FormGroup row>
+            <FormControl>
+                <InputLabel htmlFor={componentId}>
+                    {isCrib ? 'Crib' : 'Hand'} Points
+                </InputLabel>
+                <Input
+                    disabled={!isCrib && !isBlocking}
+                    id={componentId}
+                    type='number'
+                    onChange={event => {
+                        setPoints(Number(event.target.value));
+                    }}
+                />
+            </FormControl>
+            <Button
+                disabled={!isBlocking || points < 0}
+                variant='contained'
+                color='primary'
+                endIcon={<SendIcon />}
+                onClick={() => submitAction({ points })}
```

Comment:
:mag: Should this include `setPoints(0);`?

Reply:
Well, maybe. In the long run, submitting an action should trigger a websocket to tell the client that we've moved on to a different type of action and we should render something else. Right now, whether you leave the points or  `setPoints(0)`, submitting again will be an error on the server since it'd be a mistimed action.

Reply:
https://reactjs.org/docs/hooks-state.html
TL;DR - yes. `useState` creates state that persists across renders. In addition, the setter returned by `useState` causes the component to rerender when it's called (I think unless it's set to the same value as it currently has)